falco.yaml

```diff
@@ -304,6 +306,33 @@ outputs:
 # defined.
 rule_matching: first
 
+# [Experimental] `outputs_queue`
+#
+# Falco utilizes tbb::concurrent_bounded_queue for handling outputs, and this parameter
+# allows you to customize the queue capacity. Please refer to the official documentation:
+# https://oneapi-src.github.io/oneTBB/main/tbb_userguide/Concurrent_Queue_Classes.html.
+# On a healthy system with optimized Falco rules, the queue should not fill up.
+# If it does, it is most likely happening due to the entire event flow being too slow,
+# indicating that the server is under heavy load.
+# 
+# Lowering the number of items can prevent memory from steadily increasing until the OOM
+# killer stops the Falco process. We provide recovery actions to self-limit or self-kill
+# in order to handle this situation earlier, similar to how we expose the kernel buffer size
+# as a parameter.
+# However, it will not address the root cause of the event pipe not keeping up.
+# 
+# `capacity`: the maximum number of items allowed in the queue, defaulting to 0. This means that
+# the queue remains unbounded aka this setting is disabled.
+# You can experiment with values greater or smaller than the anchor value 1000000.
+# 
+# `recovery`: the strategy to follow when the queue becomes filled up. This also applies when
```

Comment:
I agree with @Andreagit97 

However, the message says:
> This also applies when the queue is unbounded, and all available memory on the system is consumed.

@incertum Can you elaborate more on how this recovery strategy works when the queue is unbounded?
:thinking:

Reply:
btw,

> try_push(item) pushes item only if it would not exceed the queue’s capacity.

So, in case of an unbound queue, `try_push` will never return false, if I understood correctly.

Reply:
yes until OOM and at that point all was lost anyways in the past.